packages/components/src/drop-zone/animators/maybe-fade.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { useReducedMotion } from '@wordpress/compose';
+
+/**
+ * Handles fade-in/fade-out animation for its children if `reducedMotion` is disabled,
+ * if not, just shows/hides the children without any animation.
+ *
+ * Relies on the animations being defined in CSS styles.
+ */
+export const MaybeFade: React.FC< {
+	show: boolean;
+	children: ReactNode;
+	duration?: number;
+} > = ( { show, children } ) => {
+	const [ shouldRender, setRender ] = useState( show );
+	const reducedMotion = useReducedMotion();
```

Comment:
Instead of triggering rerenders if the user changes their reduced motion preference, we could just do it in CSS, which if we end up using transitions as I recommend below, we could do with something as simple as:

```
@media ( prefers-reduced-motion: reduce ) {
	transition: none;
}
```

Reply:
Oh, I had no idea there was a media query for that. Thanks for the suggestion! 👍🏻